### PR TITLE
[UNDERTOW-1816] HTTPS connection abruptly closed by HttpServerConnection

### PR DESCRIPTION
**Pull request by spyrkob**
Issue: https://issues.redhat.com/browse/UNDERTOW-1816

### CODE REVIEW

File: servlet/src/test/java/io/undertow/servlet/test/handlers/AbstractHttpContinueServletTestCase.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/handlers/AbstractHttpContinueServletTestCase.java:3]: it should be 2020

File: servlet/src/test/java/io/undertow/servlet/test/handlers/HttpContinueSslServletTestCase.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline servlet/src/test/java/io/undertow/servlet/test/handlers/HttpContinueSslServletTestCase.java:3]: same here

**Review by fl4via — CHANGES_REQUESTED** [review]

**Review by fl4via — APPROVED** [review]

**Comment by fl4via** [comment]
@spyrkob  it appears that the fix is causing the tests to loop

**Comment by fl4via** [comment]
@spyrkob I ran this in my system. The HttpContinueConduitWrappingHandlerBufferLeakTestCase is looping or freezing.

**Comment by fl4via** [comment]
This is the test that is freezing. It is trying to read a response and is not getting any answer: HttpContinueConduitWrappingHandlerBufferLeakTestCase.testEmptySSLHttpContinueNoLeak:

`"main" #1 prio=5 os_prio=0 tid=0x00007effc400a800 nid=0x3b7f runnable [0x00007effc8aeb000]
   java.lang.Thread.State: RUNNABLE
        at java.net.SocketInputStream.socketRead0(Native Method)
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
        at java.net.SocketInputStream.read(SocketInputStream.java:171)
        at java.net.SocketInputStream.read(SocketInputStream.java:141)
        at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
        at sun.security.ssl.InputRecord.read(InputRecord.java:503)
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:990)
        - locked <0x00000000fbd4ca50> (a java.lang.Object)
        at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:948)
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
        - locked <0x00000000fbd4cb30> (a sun.security.ssl.AppInputStream)
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:71)
        - locked <0x00000000fbd4cb30> (a sun.security.ssl.AppInputStream)
        at io.undertow.server.handlers.HttpContinueConduitWrappingHandlerBufferLeakTestCase.testEmptySSLHttpContinueNoLeak(HttpContinueConduitWrappingHandlerBufferLeakTestCase.java:118)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
`

**Comment by spyrkob** [comment]
@fl4via the hanging test can be fixed by changing the request to HTTP/2.0 rather than HTTP/1.1 so that upgrade doesn't happen. But there's also another issue - it seems with http2 enabled terminating request in HttpContinueReadHandler is not enough and HttpContinueConduitWrappingHandlerBufferLeakTestCase#testHttpContinueRejectedBodySentAnywayNoBufferLeak still fails.

**Comment by fl4via** [comment]
@spyrkob  I'll be afk in the next couple of hours, so feel free to investigate this one if you want. I'll look into this early in the afternoon, after a team call I have scheduled for noon. If you find anything, ping me or post it here so I can pick up from where you left :-) Thanks!

**Comment by spyrkob** [comment]
I can't reproduce this fails in my env. I don't know if it's worth trying to run the CU again?

**Pull request merged**